data_validation/validation_builder.py

```diff
@@ -397,37 +380,21 @@ def get_target_query(self):
             "data_client": self.target_client,
             "schema_name": self.config_manager.target_schema,
             "table_name": self.config_manager.target_table,
+            "target_query": self.config_manager.target_query,
         }
+
         if self.validation_type == consts.CUSTOM_QUERY:
-            target_input_query = self.get_query_from_file(
-                self.config_manager.target_query_file[0]
+            table = clients.get_ibis_query(
```

Comment:
line 361-364 in config_manager.py fetches the table object. Do we need this again here?

Reply:
This is correct for normal as well as custom-query validations.
We already have the source (and target) ibis table expression generated in `config_manager.py`, and then we calculate it again in `validation_builder.py`.
This is because we are not adding source (and target) table expressions in the ConfigManager object (function `build_config_manager()` in `config_manager.py`)

@nehanene15 Let us know if this is correct understanding, and if we want to tackle this now or maybe at a later stage

Reply:
We should keep this as clients.get_ibis_query since config manager methods are source/target specific i.e. get_source_ibis_table()

In the current query_builder we also use clients.get_ibis_query so we can keep this logic for now.

Reply:
Actually, after looking at this further we should use the config_manager for this - we should do table = self.config_manager. get_target_ibis_table_from_query()

Reply:
Agreed. This is incorporated [here](https://github.com/GoogleCloudPlatform/professional-services-data-validator/pull/607/commits/f0eb38bd9e472989bcb8a8ed43a69350c1d28973)